### PR TITLE
Make the type checker more strict

### PR DESCRIPTION
**Pull request by aidantwoods**
In hindsight allowing `null` by default probably isn't a good idea, so extend the syntax to support nullables, and make changes so that where nullables are needed information is explicitly conveyed to the type checker.

I don't think this should have any BC breaks (since the type checker isn't part of the public API), and nullables in the public API are documented where they are intentional.

### CODE REVIEW

**Comment by aidantwoods** [comment]
The fact that (sans doc related) tests didn't fail in e3fffe3 means that some of the public API still needs coverage.

File: docs/v2/sameSiteCookies.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## Description
 ```php
-void sameSiteCookies ([ string $mode = null ] )
+void sameSiteCookies ([ ?string $mode = null ] )
```
Comment (aidantwoods) [inline docs/v2/sameSiteCookies.md:3]: For visibility, the only docs change was the above instance where `$mode` was nullable, has a `null` default, but wasn't explicitly marked as nullable.

**Comment by aidantwoods** [comment]
Changing milestone to a patch release since better checking for the documented types is a bugfix

**Comment by franzliedke** [comment]
I like the idea, sounds sensible. :+1:

**Pull request merged**